### PR TITLE
[RS][DDCE-623] Upgrade to Play 2.6

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-623

## Enhancement

Updated Play to 2.6. Updated SBT to 1.3.10. Updated Scala to 2.12.11. 

Updated from mockito-all to mockito-scala.

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date
 
## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/FrontendModule.scala [hunk 1]
```diff
@@ -14,87 +14,22 @@
...
 import play.api.{Configuration, Environment}
-import uk.gov.hmrc.http.{HttpGet, HttpPatch, HttpPost, HttpPut}
 import uk.gov.hmrc.play.bootstrap.http.{DefaultHttpClient, HttpClient}
-import uk.gov.hmrc.play.config.ServicesConfig
-
-class FrontendModule(val environment: Environment, val configuration: Configuration) extends AbstractModule with ServicesConfig {
 
-  override val runModeConfiguration: Configuration = configuration
-  override protected def mode = environment.mode
+class FrontendModule (val environment: Environment, val configuration: Configuration) extends AbstractModule {
```
Comment (Ufuoma-Daniel) [inline app/FrontendModule.scala:23]: Any particular reason you chose to extend AbstractModule vs Module? Just curious.

Comment (ericohwotu) [reply app/FrontendModule.scala:23]: Was already extended already

File: app/config/AppConfig.scala [hunk 2]
```diff
@@ -36,3 +47,36 @@ trait AppConfig {
   val eyaWhatsThisFlag: Boolean
 }
 
+class DefaultAppConfig @Inject()(val servicesConfig: ServicesConfig, val configuration: Configuration, val environment: Environment) extends AppConfig {
```
Comment (Ufuoma-Daniel) [inline app/config/AppConfig.scala:50]: Services config should be able to replace configuration so you don't have to inject both here.

Comment (ericohwotu) [reply app/config/AppConfig.scala:50]: true. on it

File: app/connectors/TaxHistoryConnector.scala [hunk 3]
```diff
@@ -18,49 +18,50 @@ package connectors
...
-import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpReads, HttpResponse}
+import uk.gov.hmrc.play.bootstrap.http.HttpClient
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class TaxHistoryConnector @Inject()(@Named("tax-history-baseUrl") baseUrl: URL, httpGet: HttpGet) {
+class TaxHistoryConnector @Inject()(val appConfig: AppConfig, httpClient: HttpClient)(implicit ec: ExecutionContext) {
```
Comment (Ufuoma-Daniel) [inline app/connectors/TaxHistoryConnector.scala:30]: Oh that's interesting, I don't normally see the class see an implicit execution context, I assume this is so not every method has to add it as a parameter?

File: app/controllers/ClientErrorController.scala [hunk 4]
```diff
@@ -19,38 +19,37 @@ package controllers
...
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class ClientErrorController @Inject()(val citizenDetailsConnector: CitizenDetailsConnector,
+                                      val cc: MessagesControllerComponents,
                                       override val authConnector: AuthConnector,
                                       override val config: Configuration,
                                       override val env: Environment,
```
Comment (Ufuoma-Daniel) [inline app/controllers/ClientErrorController.scala:34]: I'm not sure these need to be injected anymore?

File: app/connectors/TaxHistoryConnector.scala (hunk 3 above)
Comment (ericohwotu) [reply app/connectors/TaxHistoryConnector.scala:30]: Yes. Fraser did a little dig around and no matter which method you use its the same execution context that is passed around.

File: app/controllers/ClientErrorController.scala (hunk 4 above)
Comment (ericohwotu) [reply app/controllers/ClientErrorController.scala:34]: It is its used in AuthRedirects which is extended by AgentAuth which is extended by BaseController

File: app/config/AppConfig.scala (hunk 2 above)
Comment (RafSobol) [reply app/config/AppConfig.scala]: done & pushed

File: app/controllers/ClientErrorController.scala (hunk 4 above)
Comment (Ufuoma-Daniel) [reply app/controllers/ClientErrorController.scala:34]: Hmm seems off to me but you guys didn't bring in the AuthRedirects so no worries,

**Comment by Ufuoma-Daniel** [comment]
image
I keep running out of memory trying to compile and run unit tests? Did this happen for you?

**Comment by RafSobol** [comment]
> image
> I keep running out of memory trying to compile and run unit tests? Did this happen for you?

I think @ericohwotu was having some trouble with that - not sure how he fixed it? I couldn't reproduce it

**Pull request merged**